projects/batfish/src/main/java/org/batfish/main/Batfish.java

```diff
@@ -2121,56 +2122,16 @@ public void initBgpOriginationSpaceExplicit(Map<String, Configuration> configura
   }
 
   public InitInfoAnswerElement initInfo(boolean summary, boolean verboseError) {
-    InitInfoAnswerElement answerElement = new InitInfoAnswerElement();
     ParseVendorConfigurationAnswerElement parseAnswer = loadParseVendorConfigurationAnswerElement();
-    ConvertConfigurationAnswerElement convertAnswer = loadConvertConfigurationAnswerElement();
-    if (!summary) {
-      if (verboseError) {
-        SortedMap<String, Set<BatfishStackTrace>> errors = answerElement.getErrors();
-        parseAnswer
-            .getErrors()
-            .forEach(
-                (hostname, parseErrors) -> {
-                  errors.computeIfAbsent(hostname, k -> new HashSet<>()).add(parseErrors);
-                });
-        convertAnswer
-            .getErrors()
-            .forEach(
-                (hostname, convertErrors) -> {
-                  errors.computeIfAbsent(hostname, k -> new HashSet<>()).add(convertErrors);
-                });
-      }
-      SortedMap<String, org.batfish.common.Warnings> warnings = answerElement.getWarnings();
-      warnings.putAll(parseAnswer.getWarnings());
-      convertAnswer
-          .getWarnings()
-          .forEach(
-              (hostname, convertWarnings) -> {
-                org.batfish.common.Warnings combined = warnings.get(hostname);
-                if (combined == null) {
-                  warnings.put(hostname, convertWarnings);
-                } else {
-                  combined.getPedanticWarnings().addAll(convertWarnings.getPedanticWarnings());
-                  combined.getRedFlagWarnings().addAll(convertWarnings.getRedFlagWarnings());
-                  combined
-                      .getUnimplementedWarnings()
-                      .addAll(convertWarnings.getUnimplementedWarnings());
-                }
-              });
-    }
-    answerElement.setParseStatus(parseAnswer.getParseStatus());
-    answerElement.setParseTrees(parseAnswer.getParseTrees());
-    for (String failed : convertAnswer.getFailed()) {
-      answerElement.getParseStatus().put(failed, ParseStatus.FAILED);
-    }
+    InitInfoAnswerElement answerElement = initInfo(summary, verboseError, parseAnswer);
+    mergeConvertAnswer(summary, verboseError, answerElement);
     _logger.info(answerElement.prettyPrint());
     return answerElement;
   }
 
-  public InitInfoAnswerElement initInfoBgpAdvertisements(boolean summary, boolean verboseError) {
+  private InitInfoAnswerElement initInfo(
+      boolean summary, boolean verboseError, ParseAnswerElement parseAnswer) {
     InitInfoAnswerElement answerElement = new InitInfoAnswerElement();
-    ParseEnvironmentBgpTablesAnswerElement parseAnswer =
-        loadParseEnvironmentBgpTablesAnswerElement();
     if (!summary) {
       if (verboseError) {
         SortedMap<String, Set<BatfishStackTrace>> errors = answerElement.getErrors();
```

Comment:
I was thinking it may be useful to have a private static helper for merging errors alone: a pattern that occurs 3x in this file (and maybe elsewhere)?

(Lines 2138,2144, and and 2723).